tests/unit/resources/test_fcsfile.py

```diff
@@ -79,6 +81,32 @@ def test_gets_file_internal_compensation(ENDPOINT_BASE, client, fcs_files, spill
     assert type(comp) == Compensation
 
 
+@responses.activate
+def test_throws_correct_error_when_no_file_internal_compensation(
+    ENDPOINT_BASE, client, fcs_files
+):
+    # Given: An FcsFile with no spill string
+    file_data = fcs_files[0]
+    file_data["spillString"] = None
+    file_data["hasFileInternalComp"] = False
+    file = FcsFile.from_dict(file_data)
+    expected_response = fcs_files[0].copy()
```

Comment:
Python noob question: should the `.copy()` line come before line 89 so you're not modifying the file for all other tests?

Reply:
I _was_ modifying it for the other tests, but the fix was adding `.copy()` to the `file_data == ...` line. To be safe, I've made both `expected_response` and `file_data` copies of the original data.

Another fix for this is to change the scope of the `fcs_files` fixture to "function", which would slow down tests slightly, but recreate the fixture for each test.